src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -72,9 +70,9 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
       .write
       .insertInto("filesSource")
 
-    val dfWithTestDirectory = retryWhile[Array[Row]](
-      spark.sql(s"select * from filesSource where assetIds = Array(1739842716040355)").collect,
-      rows => rows.length < 10)
+    val dfWithTestDirectory = retryWhile[List[File]](
+      writeClient.files.filter(FilesFilter(assetIds = Some(Seq(1739842716040355L)))).compile.toList,
```

Comment:
Is it slow because we don't do pushdown on this, or because Spark overhead is large?

Reply:
This was slow because `files/list` was slow. We don't have pushdowns for Files metadata, but filter just happened to be a lot faster than list and this was the easiest option.